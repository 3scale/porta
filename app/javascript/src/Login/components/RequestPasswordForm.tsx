import { useState } from 'react'
import {
  ActionGroup,
  Button,
  Form
} from '@patternfly/react-core'

import { EmailField } from 'Login/components/FormGroups'
import { HiddenInputs } from 'Login/components/HiddenInputs'
import { validateSingleField } from 'Login/utils/formValidation'

import type { FunctionComponent } from 'react'

interface Props {
  providerLoginPath: string;
  providerPasswordPath: string;
}

interface State {
  email: string;
  validation: {
    email?: boolean;
  };
}

const RequestPasswordForm: FunctionComponent<Props> = (props) => {
  const [state, setState] = useState<State>({
    email: '',
    validation: {
      email: undefined
    }
  })

  const handleTextInputEmail: (text: string, event: React.SyntheticEvent<HTMLInputElement>) => void = (email, event) => {
    const isValid = validateSingleField(event)
    setState({ email, validation: { email: isValid } })
  }

  const emailInputProps = {
    isRequired: true,
    name: 'email',
    fieldId: 'email',
    label: 'Email address',
    isValid: state.validation.email,
    value: state.email,
    onChange: handleTextInputEmail,
    autoFocus: true
  } as const
  const formDisabled = Object.values(state.validation).some(value => !value)
  return (
    <Form
      noValidate
      acceptCharset="UTF-8"
      action={props.providerPasswordPath}
      id="request_password"
      method="post"
    >
      <HiddenInputs isPasswordReset />
      <EmailField inputProps={emailInputProps} />
      <ActionGroup>
        <Button
          className="pf-c-button pf-m-primary pf-m-block"
          isDisabled={formDisabled}
          type="submit"
        >
            Reset password
        </Button>
        <a href={props.providerLoginPath}>Sign in</a>
      </ActionGroup>
    </Form>
  )
}

export type { Props }
export { RequestPasswordForm }
