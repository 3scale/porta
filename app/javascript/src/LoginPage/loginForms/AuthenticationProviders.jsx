// @flow

import * as React from 'react'
import { KeyIcon, LessThanIcon, GreaterThanIcon } from '@patternfly/react-icons'

export type ProvidersProps = {
  authorizeURL: string,
  humanKind: string
}

type Props = {
  authenticationProviders: Array<ProvidersProps>
}

const Provider = ({authorizeURL, humanKind}: {authorizeURL: string, humanKind: string}) => {
  return (
    <p className='login-provider'>
      <a className='login-provider-link' href={authorizeURL}>
        <KeyIcon/>{' Authenticate through '}
        <LessThanIcon/><GreaterThanIcon/>{humanKind}
      </a>
    </p>
  )
}

const AuthenticationProviders = (props: Props): React.Node => {
  const {authenticationProviders} = props
  const providersList = authenticationProviders.map(
    provider => <Provider authorizeURL={provider.authorizeURL} humanKind={provider.humanKind} key={provider.humanKind}/>
  )

  return (
    <React.Fragment>
      <div className='providers-list'>
        {providersList}
      </div>
    </React.Fragment>
  )
}

export {AuthenticationProviders}
