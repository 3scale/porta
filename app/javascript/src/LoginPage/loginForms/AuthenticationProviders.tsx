import { GreaterThanIcon, KeyIcon, LessThanIcon } from '@patternfly/react-icons'

import type { FunctionComponent } from 'react'

export type ProvidersProps = {
  authorizeURL: string,
  humanKind: string
}

type Props = {
  authenticationProviders: Array<ProvidersProps>
}

const Provider: FunctionComponent<ProvidersProps> = ({ authorizeURL, humanKind }) => (
  <p className='login-provider'>
    <a className='login-provider-link' href={authorizeURL}>
      <KeyIcon />{' Authenticate through '}
      <LessThanIcon /><GreaterThanIcon />{humanKind}
    </a>
  </p>
)

// eslint-disable-next-line react/no-multi-comp
const AuthenticationProviders: FunctionComponent<Props> = (props) => {
  const { authenticationProviders } = props
  const providersList = authenticationProviders.map(
    provider => <Provider key={provider.humanKind} authorizeURL={provider.authorizeURL} humanKind={provider.humanKind} />
  )

  return (
    <div className='providers-list'>
      {providersList}
    </div>
  )
}

export { AuthenticationProviders, Props }
