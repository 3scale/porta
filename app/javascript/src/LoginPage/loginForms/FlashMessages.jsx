// @flow

import * as React from 'react'
import { ExclamationCircleIcon } from '@patternfly/react-icons'
import type {FlashMessage} from 'Types'

const FlashMessages = ({flashMessages}: {flashMessages: Array<FlashMessage>}): React.Node => {
  const messagesList = flashMessages.map(message =>
    <p key={message.message}
      className={`pf-c-form__helper-text pf-m-${message.type}`}
    >
      <ExclamationCircleIcon />
      {` ${message.message}`}
    </p>
  )
  return (
    <div>
      {messagesList}
    </div>
  )
}

export {FlashMessages}
