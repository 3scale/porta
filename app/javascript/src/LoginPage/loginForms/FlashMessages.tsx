import { ExclamationCircleIcon } from '@patternfly/react-icons'
import type { FlashMessage } from 'Types'

const FlashMessages = (
  {
    flashMessages
  }: {
    flashMessages: Array<FlashMessage>
  }
): React.ReactElement => {
  const messagesList = flashMessages.map(message =>
    <p key={message.message}
      className={`pf-c-form__helper-text pf-m-${message.type}`}
    >
      <ExclamationCircleIcon />
      {` ${message.message}`}
    </p>
  )
  return (
    <div>
      {messagesList}
    </div>
  )
}

export { FlashMessages }
