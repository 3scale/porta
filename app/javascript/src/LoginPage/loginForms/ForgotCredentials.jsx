// @flow

import * as React from 'react'
import {LoginMainFooterBandItem} from '@patternfly/react-core'

const ForgotCredentials = ({requestPasswordResetPath}: {requestPasswordResetPath: string}): React.Node =>
  <LoginMainFooterBandItem>
    <a href={requestPasswordResetPath}>Forgot password?</a>
  </LoginMainFooterBandItem>

export {ForgotCredentials}
