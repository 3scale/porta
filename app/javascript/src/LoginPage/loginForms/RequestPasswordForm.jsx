// @flow

import React from 'react'
import type { Node } from 'react'

import {
  HiddenInputs,
  EmailField,
  validateSingleField
} from 'LoginPage'

import {
  Form,
  ActionGroup,
  Button
} from '@patternfly/react-core'

type Props = {
  providerLoginPath: string,
  providerPasswordPath: string
}

type State = {
  email: string,
  validation: {
    email?: boolean
  }
}

class RequestPasswordForm extends React.Component<Props, State> {
  state: State = {
    email: '',
    validation: {
      email: undefined
    }
  }

  handleTextInputEmail (email: string, event: SyntheticEvent<HTMLInputElement>) {
    const isValid = validateSingleField(event)
    this.setState({ email, validation: {email: isValid} })
  }

  render (): Node {
    const emailInputProps = {
      isRequired: true,
      name: 'email',
      fieldId: 'email',
      label: 'Email address',
      isValid: this.state.validation.email,
      value: this.state.email,
      onChange: this.handleTextInputEmail,
      autoFocus: 'autoFocus'
    }
    const formDisabled = Object.values(this.state.validation).some(value => value !== true)
    return (
      <Form noValidate id='request_password'
        action={this.props.providerPasswordPath}
        id='request_password'
        acceptCharset='UTF-8'
        method='post'
      >
        <HiddenInputs isPasswordReset />
        <EmailField inputProps={emailInputProps}/>
        <ActionGroup>
          <Button className='pf-c-button pf-m-primary pf-m-block'
            type='submit'
            isDisabled={formDisabled}
          >Reset password</Button>
          <a href={this.props.providerLoginPath}>Sign in</a>
        </ActionGroup>
      </Form>
    )
  }
}

export {
  RequestPasswordForm
}
