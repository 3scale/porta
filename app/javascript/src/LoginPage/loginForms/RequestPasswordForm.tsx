import React from 'react'
import type { ReactNode } from 'react'

import {
  HiddenInputs,
  EmailField,
  validateSingleField
} from 'LoginPage'

import {
  Form,
  ActionGroup,
  Button
} from '@patternfly/react-core'

export type Props = {
  providerLoginPath: string,
  providerPasswordPath: string
};

type State = {
  email: string,
  validation: {
    email?: boolean
  }
};

class RequestPasswordForm extends React.Component<Props, State> {
  state: State = {
    email: '',
    validation: {
      email: undefined
    }
  };

  handleTextInputEmail: (arg1: string, arg2: React.SyntheticEvent<HTMLInputElement>) => void = (email, event) => {
    const isValid = validateSingleField(event)
    this.setState({ email, validation: { email: isValid } })
  };

  render () {
    const emailInputProps = {
      isRequired: true,
      name: 'email',
      fieldId: 'email',
      label: 'Email address',
      isValid: this.state.validation.email,
      value: this.state.email,
      onChange: this.handleTextInputEmail,
      autoFocus: true
    } as const
    const formDisabled = Object.values(this.state.validation).some(value => value !== true)
    return (
      <Form
        noValidate
        action={this.props.providerPasswordPath}
        id='request_password'
        acceptCharset='UTF-8'
        method='post'
      >
        <HiddenInputs isPasswordReset />
        <EmailField inputProps={emailInputProps}/>
        <ActionGroup>
          <Button className='pf-c-button pf-m-primary pf-m-block'
            type='submit'
            isDisabled={formDisabled}
          >Reset password</Button>
          <a href={this.props.providerLoginPath}>Sign in</a>
        </ActionGroup>
      </Form>
    )
  }
}

export {
  RequestPasswordForm
}
