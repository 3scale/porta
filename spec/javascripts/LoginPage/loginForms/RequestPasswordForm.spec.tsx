import { FormEvent } from 'react'

import { mount } from 'enzyme'

import { RequestPasswordForm, Props } from 'LoginPage/loginForms/RequestPasswordForm'

const defaultProps = {
  flashMessages: [],
  providerLoginPath: 'login-path',
  providerPasswordPath: 'password-path'
}

const mountWrapper = (props: Partial<Props> = {}) => mount<RequestPasswordForm>(<RequestPasswordForm { ...{ ...defaultProps, ...props } } />)

it('should render itself', () => {
  const wrapper = mountWrapper()
  expect(wrapper).toMatchSnapshot()
})

it('should set email and validation state to true', () => {
  const event = {
    currentTarget: {
      value: 'bob@sponge.com',
      type: 'email'
    }
  } as FormEvent<HTMLInputElement>

  const wrapper = mountWrapper()
  wrapper.find('input#email').props().onChange!(event)
  expect(wrapper.state().email).toEqual('bob@sponge.com')
  expect(wrapper.state().validation.email).toEqual(true)
})

it('should set validation state to false when email is invalid', () => {
  const event = {
    currentTarget: {
      value: 'bobspongecom',
      type: 'email'
    }
  } as FormEvent<HTMLInputElement>

  const wrapper = mountWrapper()
  wrapper.find('input#email').props().onChange!(event)
  expect(wrapper.state().email).toEqual('bobspongecom')
  expect(wrapper.state().validation.email).toEqual(false)
})
