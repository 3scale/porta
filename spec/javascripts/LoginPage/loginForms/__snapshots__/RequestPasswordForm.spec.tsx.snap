// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render itself 1`] = `
<RequestPasswordForm
  flashMessages={[]}
  providerLoginPath="login-path"
  providerPasswordPath="password-path"
>
  <Form
    acceptCharset="UTF-8"
    action="password-path"
    id="request_password"
    method="post"
    noValidate={true}
  >
    <FormBase
      acceptCharset="UTF-8"
      action="password-path"
      id="request_password"
      innerRef={null}
      method="post"
      noValidate={true}
    >
      <form
        acceptCharset="UTF-8"
        action="password-path"
        className="pf-c-form"
        id="request_password"
        method="post"
        noValidate={true}
      >
        <HiddenInputs
          isPasswordReset={true}
        >
          <input
            name="utf8"
            type="hidden"
            value="âœ“"
          />
          <input
            name="_method"
            type="hidden"
            value="delete"
          />
          <CSRFToken>
            <input
              name="mocked-csrf-param"
              type="hidden"
              value="mocked-csrf-token"
            />
          </CSRFToken>
        </HiddenInputs>
        <EmailField
          inputProps={
            {
              "autoFocus": true,
              "fieldId": "email",
              "isRequired": true,
              "isValid": undefined,
              "label": "Email address",
              "name": "email",
              "onChange": [Function],
              "value": "",
            }
          }
        >
          <FormGroup
            helperTextInvalid="A valid email address is mandatory"
            inputProps={
              {
                "autoFocus": true,
                "fieldId": "email",
                "isRequired": true,
                "isValid": undefined,
                "label": "Email address",
                "name": "email",
                "onChange": [Function],
                "value": "",
              }
            }
            type="email"
          >
            <FormGroup
              autoComplete="off"
              fieldId="email"
              helperTextInvalid="A valid email address is mandatory"
              isRequired={true}
              label="Email address"
              validated="default"
            >
              <GenerateId
                prefix="pf-random-id-"
              >
                <div
                  autoComplete="off"
                  className="pf-c-form__group"
                >
                  <div
                    className="pf-c-form__group-label"
                  >
                    <label
                      className="pf-c-form__label"
                      htmlFor="email"
                    >
                      <span
                        className="pf-c-form__label-text"
                      >
                        Email address
                      </span>
                      <span
                        aria-hidden="true"
                        className="pf-c-form__label-required"
                      >
                         
                        *
                      </span>
                    </label>
                     
                  </div>
                  <div
                    className="pf-c-form__group-control"
                  >
                    <TextInput
                      autoComplete="off"
                      autoFocus={true}
                      id="email"
                      isRequired={true}
                      name="email"
                      onChange={[Function]}
                      type="email"
                      validated="default"
                      value=""
                    >
                      <TextInputBase
                        aria-label={null}
                        autoComplete="off"
                        autoFocus={true}
                        className=""
                        id="email"
                        innerRef={null}
                        isDisabled={false}
                        isIconSprite={false}
                        isLeftTruncated={false}
                        isReadOnly={false}
                        isRequired={true}
                        name="email"
                        onChange={[Function]}
                        ouiaSafe={true}
                        type="email"
                        validated="default"
                        value=""
                      >
                        <input
                          aria-invalid={false}
                          aria-label={null}
                          autoComplete="off"
                          autoFocus={true}
                          className="pf-c-form-control"
                          data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                          data-ouia-component-type="PF4/TextInput"
                          data-ouia-safe={true}
                          disabled={false}
                          id="email"
                          name="email"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={true}
                          type="email"
                          value=""
                        />
                      </TextInputBase>
                    </TextInput>
                  </div>
                </div>
              </GenerateId>
            </FormGroup>
          </FormGroup>
        </EmailField>
        <ActionGroup>
          <div
            className="pf-c-form__group pf-m-action"
          >
            <div
              className="pf-c-form__group-control"
            >
              <div
                className="pf-c-form__actions"
              >
                <Button
                  className="pf-c-button pf-m-primary pf-m-block"
                  isDisabled={true}
                  type="submit"
                >
                  <ButtonBase
                    className="pf-c-button pf-m-primary pf-m-block"
                    innerRef={null}
                    isDisabled={true}
                    type="submit"
                  >
                    <button
                      aria-disabled={true}
                      aria-label={null}
                      className="pf-c-button pf-m-primary pf-m-disabled pf-c-button pf-m-primary pf-m-block"
                      data-ouia-component-id="OUIA-Generated-Button-primary-1"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe={true}
                      disabled={true}
                      role={null}
                      tabIndex={null}
                      type="submit"
                    >
                      Reset password
                    </button>
                  </ButtonBase>
                </Button>
                <a
                  href="login-path"
                >
                  Sign in
                </a>
              </div>
            </div>
          </div>
        </ActionGroup>
      </form>
    </FormBase>
  </Form>
</RequestPasswordForm>
`;
